Tribler/community/tunnel/tunnel_community.py

```diff
@@ -1198,8 +1198,8 @@ def exit_data(self, circuit_id, sock_addr, destination, data):
         else:
             self.tunnel_logger.error("Dropping data packets with unknown circuit_id")
 
-    def crypto_out(self, circuit_id, content, is_data=False):
-        circuit = self.circuits.get(circuit_id, None)
+    def crypto_out(self, circuit, content, is_data=False):
+        circuit_id = circuit.circuit_id
```

Comment:
Unfortunately, I tried this as well. But it cannot work. Look at the if statement at line 1210